### PR TITLE
Memory fixes

### PR DESCRIPTION
**Pull request by Sadanand14**
- memory leakage brought down to 28 bytes.
- fixed projection calculation each frame.

### CODE REVIEW

File: Falcon/Falcon/Rendering/PipeLine/Renderer.cpp [hunk 1]
```diff
@@ -86,14 +92,15 @@ void Renderer::CreateDrawStates()
 /**
 *Function to Set the relevant data in the draw states.
 */
-void Renderer::SetDrawStates(boost::container::vector<Entity*, fmemory::StackSTLAllocator<Entity*>>* entities)
+void Renderer::SetDrawStates(boost::container::vector<Entity*, fmemory::StackSTLAllocator<Entity*>>* entities, glm::mat4 projection)
```
Comment (ShantanuJamble) [inline Falcon/Falcon/Rendering/PipeLine/Renderer.cpp]: Use ref instead of copy?

File: Falcon/Falcon/Rendering/PipeLine/Renderer.cpp [hunk 2]
```diff
@@ -146,48 +153,52 @@ void Renderer::SetDrawStates(boost::container::vector<Entity*, fmemory::StackSTL
...
 	m_RES->ProcessEvents();
-	glm::mat4 projection = glm::perspective(glm::radians(cam.m_Zoom), (float)width / (float)height, 0.1f, 100.0f);
-
-	shader->UseShader();
-	shader->SetMat4("projection", projection);
-	shader->SetMat4("view", cam.GetViewMatrix());
+	m_entities = entities;
+	for (unsigned int i = 0; i < m_entities->size(); ++i)
+	{
+		Shader* shader = m_entities->at(i)->GetComponent<RenderComponent>()->m_mesh->GetMaterial()->m_shader;
```
Comment (pyrophone) [inline Falcon/Falcon/Rendering/PipeLine/Renderer.cpp]: Not all entities will have the rendercomponent, so you should check if it does before accessing it's variables

File: Falcon/Falcon/Rendering/PipeLine/Renderer.cpp (hunk 1 above)
Comment (Sadanand14) [reply Falcon/Falcon/Rendering/PipeLine/Renderer.cpp]: not possible. The projection being passed is defined on the stack.

File: Falcon/Falcon/Rendering/PipeLine/Renderer.cpp (hunk 2 above)
Comment (Sadanand14) [reply Falcon/Falcon/Rendering/PipeLine/Renderer.cpp]: yeah I wondered why it wasn't crashing on visual studios without that check. Seems visual studio is effectively ruining programming.

**Pull request merged**